### PR TITLE
[TIMOB-25839] Android/iOS: Modified touch events to use default units defined in "tiapp.xml"

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-25839

**Summary:**
- This is a breaking-change.
- Android touch coordinates used to be hard-coded to pixels.
- iOS touch coordinates used to be hard-coded to dips.

**Test:**
_Run the below test on both Android and iOS._
1. Create a classic app using the test code attached to [TIMOB-25839](https://jira.appcelerator.org/browse/TIMOB-25839).
2. Set "tiapp.xml" property "ti.ui.defaultunit" to `dp`.
3. Run the project.
4. Drag your finger within the light-gray rectangle.
5. Verify that a green square is centered under your finger while you drag.
6. Set "tiapp.xml" property "ti.ui.defaultunit" to `px`.
7. Repeat steps 3-5.

### CODE REVIEW

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiUtils.m [hunk 1]
```diff
@@ -1204,9 +1204,11 @@ + (NSDictionary *)sizeToDictionary:(CGSize)size
 + (NSDictionary *)touchPropertiesToDictionary:(UITouch *)touch andView:(UIView *)view
 {
   if ([self forceTouchSupported] || [self validatePencilWithTouch:touch]) {
```
Comment (vijaysingh-axway) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiUtils.m]: In case if condition failed, it will return the old value. That should also be updated .

Comment (jquick-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiUtils.m]: Oh I see. You're talking about the `pointToDictionary()` call down below. I'll update it. Thanks.

Comment (jquick-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiUtils.m]: PR updated.

**Review by garymathews — APPROVED** [review]
CR: Android changes - **PASS**

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Review by ssjsamir — APPROVED** [review]
**FR Passed** Touch event coordinate units now match between platforms.

**Test Steps**

1. Created a new SDK build from this PR
2. Created a new titanium application 
3. Copied the test case from https://jira.appcelerator.org/browse/TIMOB-25839
4. Ran the App on Android setting "tiapp.xml" property "ti.ui.defaultunit" to `dp`
5. Dragged my finger across the screen 
6. Could see the green square under my finger at all times 
7. Repeated the above test but this time setting the "tiapp.xml" property "ti.ui.defaultunit" to `px`
8. Able to verify the same results 
**Followed the same steps for iOS.** 

**Test Environment**
```
APPC Studio: 5.1.0.201808080937
iPhone 7 Sim (iOS 12.1)
Pixel XL Device (9.0)
APPC CLI: 7.0.8
Operating System Name: Mac OS Mojave
Operating System Version: 10.14.1
Node.js Version: 8.9.1
Xcode 10.1
```

**Pull request merged**